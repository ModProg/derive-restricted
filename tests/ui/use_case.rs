use derive_where::derive_where;

#[derive_where(Clone)]
struct StructEmpty {}

#[derive_where(Clone)]
struct TupleEmpty();

#[derive_where(Clone)]
struct Unit;

#[derive_where(Clone)]
struct UnionEmpty {}

#[derive_where(Clone)]
struct StructNone(u8);

#[derive_where(Clone)]
enum EnumNone {
	A(u8),
}

#[derive_where(Clone; T)]
struct SameGenerics<T>(T);

#[derive_where(Clone, Debug; T)]
#[derive_where(skip_inner)]
struct Skip<T>(T);

#[derive_where(Clone, Default; T)]
enum Default<T> {
	#[derive_where(default)]
	A(T),
}

fn main() {}
