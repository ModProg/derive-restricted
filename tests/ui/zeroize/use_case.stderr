error: this can be handled by standard `#[derive(..)]`, use a `skip` or `incomparable` attribute, implement `Default` on an enum, or different generic type parameters
 --> tests/ui/zeroize/use_case.rs:5:16
  |
5 | #[derive_where(Clone, Zeroize(crate = zeroize_); T)]
  |                ^^^^^

error: this can be handled by standard `#[derive(..)]`, use a `skip` or `incomparable` attribute, implement `Default` on an enum, or different generic type parameters
 --> tests/ui/zeroize/use_case.rs:8:16
  |
8 | #[derive_where(Clone, Zeroize; T)]
  |                ^^^^^
